### DISCUSSION TITLE
Support 2-blank-line-termination in REPL in Scala 3

### DISCUSSION THREAD

**Discussion by cruiztorresj** [scala/scala3#17712] [Feature Requests]
Hello, I was trying to terminate a statement pressing enter twice but it only works for Scala 2, can you please make it available for Scala 3 over Java 17 too?

Here's a screenshot depicting the issue both with SBT and the **Scala 3** standalone application.
![console3](https://user-images.githubusercontent.com/75508473/178163843-d0e93c4b-44c3-4506-b118-a4917f9bf2be.png)

And here's a **successful** execution within **Scala 2**.
![console2](https://user-images.githubusercontent.com/75508473/178163836-e6feef5f-108d-4f66-872b-82e1e9f34489.png)

Java Virtual Machine Information:
`java 17.0.3.1 2022-04-22 LTS`
`Java(TM) SE Runtime Environment (build 17.0.3.1+2-LTS-6)`
`Java HotSpot(TM) 64-Bit Server VM (build 17.0.3.1+2-LTS-6, mixed mode, sharing)`

Thanks!

--
Caleb.

**Comment by som-snytt** [comment]
I think this was deemed unnecessary because of magic JLine editing capabilities, but I can't find the ticket or conversation. I also thought ctl-c was supported, but for me it exits the REPL.

**Comment by cruiztorresj** [comment]
Thanks for getting back with an answer.
I am not sure why it was deemed that way, IMO it is a Scala's REPL core functionality. It's even promoted in some books, e.g. THE [Scala book](https://www.artima.com/shop/programming_in_scala_5ed) by Martin Odersky et all.
Have a good day!

--
Caleb

**Comment by bishabosha** [comment]
If I could guess why this occurs it would be to support optional braces syntax - however to see 7 blank lines before any code is a bit absurd :) 

Edit: I mean absurd for REPL to accept 7 blank lines as intentional code

**Comment by cruiztorresj** [comment]
Not absurd at all, those seven blank lines were meant to showcase _enter_ was hit n (Seven in this case) number of times without the REPL terminating the statement as intended.
Thanks for changing this issue title, yours is more adequate.

Well, good luck buds. 🚀

**Comment by SethTisue** [comment]
Scala 2 lost this behavior around the time the JLine 3 upgrade landed (https://github.com/scala/bug/issues/12266), but Dale and I restored it in https://github.com/scala/scala/pull/9407

as discussed at 12266, control-G also works for noping out (including in Scala 3), so that's a workaround for now though as I wrote there:

> tends to leave junk on the screen, but perhaps that's JLine's fault rather than something in the Scala integration
